### PR TITLE
chore(cts): partial assert on e2e

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket: https://algolia.atlassian.net/browse/DI-1736

### Changes included:

Because some values of the API can vary, we can change the e2e tests a bit to only assert the fields we want. e.g. for the `search` method, `processingTimeMS` can vary, so we remove it from the expected body of the CTS and only assert the remaining fields.

I did not found native ways to do it in PHP, JavaScript and Python, so I wrote a small helper

### CODE REVIEW

File: templates/php/tests/requests/requests.mustache [hunk 1]
```diff
@@ -36,6 +36,33 @@ class {{clientPrefix}}Test extends TestCase implements HttpClientInterface
      */
     private $recordedRequests = [];
 
+    {{#hasE2E}}
+    protected function union($expected, $received)
```
Comment (shortcuts) [inline templates/php/tests/requests/requests.mustache:40]: idk how namespaces etc. works so it's inlined for now, there's a loooot of duplicated stuff in the PHP test (e.g. `assertRequests`) etc. would be great it we move all of that to an helper file cc @damcou

**Comment by Fluf22** [comment]
🆗 
It would mean one more manual helper to do in each language?
We definitely need a list of stuff required to do when adding a new language

**Review by Fluf22 — DISMISSED** [review]

**Comment by shortcuts** [comment]
> 🆗 It would mean one more manual helper to do in each language?

In the tests yes only if there's no native way to do so, however since e2e will be required at some point, we'll know right away but it's a one time write

> We definitely need a list of stuff required to do when adding a new language

I'm currently writing a small doc that lists helpers, where they live and what they do

**Comment by shortcuts** [comment]
great example of the necessity of this addition: https://github.com/algolia/api-clients-automation/actions/runs/7337323488/job/19978111243?pr=2442

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -261,7 +261,7 @@ jobs:
         run: cd clients/algoliasearch-client-javascript && yarn workspace algoliasearch test
 
       - name: Remove previous CTS output
-        run: rm -rf tests/output/javascript/src || true
+        run: rm -rf tests/output/javascript/src/requests tests/output/javascript/src/client || true
```
Comment (millotp) [inline .github/workflows/check.yml:264]: maybe we could use `fromJSON(needs.setup.outputs.JAVASCRIPT_DATA).testsToDelete`, this was just a simplification

File: templates/python/tests/requests/requests.mustache [hunk 3]
```diff
@@ -4,6 +4,7 @@ from algoliasearch.http.transporter import EchoTransporter
 from algoliasearch.{{{import}}}.client import {{#lambda.pascalcase}}{{{client}}}{{/lambda.pascalcase}}
 from algoliasearch.{{{import}}}.config import Config
 {{#hasE2E}}
+from ..helpers import Helpers
```
Comment (millotp) [inline templates/python/tests/requests/requests.mustache:7]: never saw that syntax before !

**Review by millotp — COMMENTED** [review]
this is very nice !

File: templates/python/tests/requests/requests.mustache (hunk 3 above)
Comment (shortcuts) [reply templates/python/tests/requests/requests.mustache:7]: python is full of different syntax.... that's confusing ahah

File: .github/workflows/check.yml (hunk 2 above)
Comment (shortcuts) [reply .github/workflows/check.yml]: ah yes I forgot about that good idea

Comment (millotp) [reply .github/workflows/check.yml]: hope it works without the `|| true`

**Review by millotp — APPROVED** [review]

**Pull request merged**